src/FffGcodeWriter.cpp

```diff
@@ -2496,7 +2499,8 @@ void FffGcodeWriter::setExtruder_addPrime(const SliceDataStorage& storage, Layer
     }
 
     const unsigned int previous_extruder = gcode_layer.getExtruder();
-    if (previous_extruder == static_cast<unsigned int>(extruder_nr) && !(static_cast<unsigned int>(extruder_nr) == outermost_prime_tower_extruder && gcode_layer.getLayerNr() >= -Raft::getFillerLayerCount(storage))) //No unnecessary switches, unless switching to extruder for the outer shell of the prime tower.
+    if (previous_extruder == static_cast<size_t>(extruder_nr)
```

Comment:
We can't do that, since this variable can actually be -1 (see several lines higher where it checks for that).

Reply:
All the places where this function is called, the `extruder_nr` parameter is always defined as `size_t`. So I guess it will never be a negative number.